### ISSUE TITLE
STM32F401 Nucleo-64 no longer programmable

### ISSUE THREAD

**Issue by Quadricopter** [texane/stlink#766]
Since commit 7651d2116fd74c7803ea00ab1da7cf3d00faf44c, I cannot flash my Nucleo-64 anymore :-(
```
> st-flash write nucleo.bin 0x8000000
st-flash 1.5.1-12-g30de1b3-dirty
2019-01-28T14:03:16 INFO common.c: Loading device parameters....
2019-01-28T14:03:16 INFO common.c: Device connected is: F4 device (Dynamic Efficency), id 0x10016433
2019-01-28T14:03:16 INFO common.c: SRAM size: 0x18000 bytes (96 KiB), Flash: 0x80000 bytes (512 KiB) in pages of 16384 bytes
2019-01-28T14:03:16 INFO common.c: Attempting to write 4020 (0xfb4) bytes to stm32 address: 134217728 (0x8000000)
Flash page at addr: 0x08000000 erased
2019-01-28T14:03:16 INFO common.c: Finished erasing 1 pages of 16384 (0x4000) bytes
2019-01-28T14:03:16 INFO common.c: Starting Flash write for F2/F4/L4
2019-01-28T14:03:16 INFO flash_loader.c: Successfully loaded flash loader in sram
enabling 32-bit flash writes
size: 4020
2019-01-28T14:03:19 ERROR flash_loader.c: flash loader run error
2019-01-28T14:03:19 ERROR common.c: stlink_flash_loader_run(0x8000000) failed! == -1
stlink_fwrite_flash() == -1
Makefile:44: recipe for target 'flash' failed
make: *** [flash] Error 255
```

Comment following line makes st-flash work again with Nucleo-64
https://github.com/texane/stlink/blob/30de1b30e7a96974ae764d0508c151eadb1addd1/src/flash_loader.c#L265

```
> lsusb
Bus 001 Device 024: ID 0483:374b STMicroelectronics ST-LINK/V2.1 (Nucleo-F103RB)
```

**Comment by xor-gate** [comment]
We should continue in #761 because this is duplicate. We should skim through the PRs which broke the flashing.

**Comment by Nightwalker-87** [comment]
Fixed in commits f5d0454ab0a18f209ad5b7d0d51f3945b27dd892 and b40c2436b68a35544a1eae10f4b6a3de7648a566

**Comment by Nightwalker-87** [comment]
Fixed in commits f5d0454ab0a18f209ad5b7d0d51f3945b27dd892 and b40c2436b68a35544a1eae10f4b6a3de7648a566.

**Comment by Quadricopter** [comment]
Hello,

tested with master branch ( commit: f5d0454ab0a18f209ad5b7d0d51f3945b27dd892 )
**It's OK for me!**

> st-flash write nucleo.bin 0x8000000
st-flash 1.6.0-31-gf5d0454
2020-03-24T19:46:31 INFO common.c: Loading device parameters....
2020-03-24T19:46:31 INFO common.c: Device connected is: F4 device (Dynamic Efficency), id 0x10016433
2020-03-24T19:46:31 INFO common.c: SRAM size: 0x18000 bytes (96 KiB), Flash: 0x80000 bytes (512 KiB) in pages of 16384 bytes
2020-03-24T19:46:31 INFO common.c: Attempting to write 12368 (0x3050) bytes to stm32 address: 134217728 (0x8000000)
Flash page at addr: 0x08000000 erased
2020-03-24T19:46:32 INFO common.c: Finished erasing 1 pages of 16384 (0x4000) bytes
2020-03-24T19:46:32 INFO common.c: Starting Flash write for F2/F4/L4
2020-03-24T19:46:32 INFO flash_loader.c: Successfully loaded flash loader in sram
enabling 32-bit flash writes
size: 12368
2020-03-24T19:46:32 INFO common.c: Starting verification of write complete
2020-03-24T19:46:32 INFO common.c: Flash written and verified! jolly good!
